tests/components/sensor/test_mqtt.py

```diff
@@ -37,11 +38,49 @@ def test_setting_sensor_value_via_mqtt_message(self):
         fire_mqtt_message(self.hass, 'test-topic', '100')
         self.hass.block_till_done()
         state = self.hass.states.get('sensor.test')
-
+        
         self.assertEqual('100', state.state)
         self.assertEqual('fav unit',
                          state.attributes.get('unit_of_measurement'))
 
+    def test_setting_sensor_value_expires(self):
+        """Test the expiration of the value."""
+        mock_component(self.hass, 'mqtt')
+        assert setup_component(self.hass, sensor.DOMAIN, {
+            sensor.DOMAIN: {
+                'platform': 'mqtt',
+                'name': 'test',
+                'state_topic': 'test-topic',
+                'unit_of_measurement': 'fav unit',
+                'expire_after': '2',
+                'force_update': True
+            }
+        })
+
+        state = self.hass.states.get('sensor.test')
+        self.assertEqual('unknown', state.state)
+
+        fire_mqtt_message(self.hass, 'test-topic', '100')
+        self.hass.block_till_done()
+
+        state = self.hass.states.get('sensor.test')
+        self.assertEqual('100', state.state)
+        
+        time.sleep(1)
```

Comment:
You are never allowed to call time.sleep in tests. Please use the `mock_fire_time_changed` helper.